server/apps/users/services.py

```diff
@@ -102,6 +103,29 @@ def compare_preferences(original, updates):
     return added, removed, modified
 
 
+def set_sign_off(user):
+    """
+    Set sign_off property on user if it's not set already.
+    """
+
+    if SIGN_OFF not in user:
+        if 'first_name' not in user or 'last_name' not in user:
+            user[SIGN_OFF] = user['username'][:3].upper()
+        else:
+            user[SIGN_OFF] = '%s%s' % (user['first_name'][0], user['last_name'][0])
```

Comment:
I find `{first_name[0]}{last_name[0]}'.format(**user)` slightly more readable, although it's just my personal opinion, thus a change is completely at your discretion.